plugins/generic/externalFeed/ExternalFeedPlugin.inc.php

```diff
@@ -1,367 +1,322 @@
+		$output =& $args[2];
+		$request =& Registry::get('request');
+		$dispatcher = $request->getDispatcher();
+		$output .= '<li><a name="externalFeeds" href="' . $dispatcher->url($request, ROUTE_COMPONENT, null, 'plugins.generic.externalFeed.controllers.grid.ExternalFeedGridHandler', 'index') . '">' . __('plugins.generic.externalFeed.displayName') . '</a></li>';
+		return false;
+	}
+	
+	/**
+	 * Permit requests to the external feeds grid handler
+	 * @param $hookName string The name of the hook being invoked
+	 * @param $args array The parameters to the invoked hook
+	 */
+	function setupGridHandler($hookName, $params) {
+		$component =& $params[0];
+		if ($component == 'plugins.generic.externalFeed.controllers.grid.ExternalFeedGridHandler') {
+			// Allow the static page grid handler to get the plugin object
+			import($component);
+			ExternalFeedGridHandler::setPlugin($this);
+			return true;
+		}
+		return false;
+	}
+	
+	/**
+	 * Get the filename of the CSS stylesheet for this plugin.
+	 */
+	public function getStyleSheetFile() {
+		$request = $this->getRequest();
+		$journal = $request->getJournal();
+		$journalId = $journal->getId();
+		
+		$styleSheet = $this->getSetting($journalId, 'externalFeedStyleSheet');
+	
+		if (empty($styleSheet)) {
+			return $this->getDefaultStyleSheetFile();
+		} else {
+			import('classes.file.PublicFileManager');
+			$fileManager = new PublicFileManager();
+			return $fileManager->getJournalFilesPath($journalId) . '/' . $styleSheet['uploadName'];
+		}
+	}
+	
+	/**
+	 * Display external feed content on journal homepage.
+	 * @param $hookName string
+	 * @param $args array
+	 */
+	public function displayHomepage($hookName, $args) {
+		$request = $this->getRequest();
+		$journal = $request->getJournal();
+		$journalId = $journal->getId();
+		
+		if ($this->getEnabled()) {
+			if (!is_a($request->getRouter(), 'PKPPageRouter')) return false;
+			$requestedPage = $request->getRequestedPage();
+			
+			$entries = array();
+			if (empty($requestedPage) || $requestedPage == 'index') {
+				$externalFeedDao =& DAORegistry::getDAO('ExternalFeedDAO');
+				$this->import('simplepie.SimplePie');
+				
+				$feeds =& $externalFeedDao->getExternalFeedsByJournalId($journal->getId());
+				while ($currentFeed =& $feeds->next()) {
+					if (!$currentFeed->getDisplayHomepage()) continue;
+					$items = null;
+					$feedTitle = null;
+					
+					$feed = new SimplePie();
+					$feed->set_feed_url($currentFeed->getUrl());
+					$feed->enable_order_by_date(false);
+					$feed->set_cache_location(CacheManager::getFileCachePath());
+					$feed->init();
+					
+					if ($currentFeed->getLimitItems()) {
+						$recentItems = $currentFeed->getRecentItems();
+					} else {
+						$recentItems = 0;
+					}
+					
+					$entries[] = array(
+						'feedTitle' => $currentFeed->getLocalizedTitle(),
+						'items' => $feed->get_items(0, $recentItems),
+					);
+				}
+				
+				$templateManager =& $args[0];

```

Comment:
Since we're now requiring PHP5 (phew!) we're reducing dependency on references (to make the code more readable, and to reduce the chances of weird, hard-to-track reference-related bugs). In PHP5, objects are automatically treated by reference, so cases like this can be simplified to remove the reference `&`.